lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -6326,11 +6322,20 @@ LogicalResult LayerBlockOp::verify() {
           if (isa<RefDefineOp>(connect))
             return WalkResult::advance();
 
-          // We can drive any destination inside the current layerblock.
+          // Verify that connects only drive values declared in the layer block.
+          // If we see a non-passive connect destination, then verify that the
+          // source is in the same layer block so that the source is not driven.
           auto dest = getFieldRefFromValue(connect.getDest()).getValue();
```

Comment:
I was concerned about this, too. Thanks for the pointer to `FieldRefCache`.

A layer region can be enormous. I expect that it is worthwhile to use the cache over that region. This verifier is already sketchy as it is verifying so much. It would be awful if hits performance issues for any reason.

Switched to `FieldRefCache` in c08b562e0.